src/Uno.UI/UI/Xaml/Input/FocusManager.wasm.cs

```diff
@@ -44,10 +47,30 @@ internal static void ProcessElementFocused(UIElement element)
 				_log.Value.LogDebug($"{nameof(ProcessElementFocused)}() focusedElement={GetFocusedElement()}, element={element}, searching for focusable parent control");
 			}
 
-			foreach (var parent in element.GetParents())
+			foreach (var candidate in element.GetAllParents())
 			{
 				// Try to find the first focusable parent and set it as focused, otherwise just keep it for reference (GetFocusedElement())
-				if (parent is TextBlock textBlock && textBlock.IsFocusable)
+
+				// Special handling for RootVisual - which is not focusable on managed side
+				// but is focusable on native side. The purpose of this trick is to allow
+				// us to recognize, that the page was focused by tabbing from the address bar
+				// and focusing the first focusable element on the page instead.
+				if (candidate is RootVisual rootVisual)
```

Comment:
@ramezgerges Can you try creating a UI test (or runtime test) that validates this particular block? Are we able to programmatically invoke this particular piece?

Reply:
I don't think so. This only gets called on `ReceivedFocusNative`. This is preexisting code by the way, I just moved it inside `ProcessElementFocused` where it makes more sense IMHO.

Reply:
Actually, after thinking about some hacks, there doesn't seem to be a straightforward way to realisticly unfocus and refocus a tab/window I think it's not that critical and it hasn't been touched in ages, so it's probably fine.

Reply:
Ok, then let's make a sample marked as a manual test which explains what to do and what to expect. Thanks!